fdbserver/worker.actor.cpp

```diff
@@ -1328,20 +1329,56 @@ ACTOR Future<Void> monitorHighMemory(int64_t threshold) {
 	return Void();
 }
 
+struct StorageDiskCleaner {
+	UID id;
+	KeyValueStoreType storeType;
+	LocalityData locality;
+	std::string filename;
+	Future<Void> future;
+};
+
 struct TrackRunningStorage {
 	UID self;
 	KeyValueStoreType storeType;
+	LocalityData locality;
+	std::string filename;
 	std::set<std::pair<UID, KeyValueStoreType>>* runningStorages;
+	std::vector<StorageDiskCleaner>* storageCleaners;
+
 	TrackRunningStorage(UID self,
 	                    KeyValueStoreType storeType,
-	                    std::set<std::pair<UID, KeyValueStoreType>>* runningStorages)
-	  : self(self), storeType(storeType), runningStorages(runningStorages) {
+	                    LocalityData locality,
+	                    const std::string& filename,
+	                    std::set<std::pair<UID, KeyValueStoreType>>* runningStorages,
+	                    std::vector<StorageDiskCleaner>* storageCleaners)
+	  : self(self), storeType(storeType), locality(locality), filename(filename), runningStorages(runningStorages),
+	    storageCleaners(storageCleaners) {
 		runningStorages->emplace(self, storeType);
 	}
-	~TrackRunningStorage() { runningStorages->erase(std::make_pair(self, storeType)); };
+	~TrackRunningStorage() {
+		runningStorages->erase(std::make_pair(self, storeType));
+
+		// Start a disk cleaner except for tss data store
+		if (filename.find(testingStoragePrefix.toString()) == std::string::npos) {
```

Comment:
Although the tss prefix is not likely to exist elsewhere in the file, for a prefix check I think you want `!= 0`.

Reply:
the file name here is the full path("/mnt/fdb/....".. so it's not start with testingStorageprefix. I could split the absolution path and pick only filename to match. but I think current check looks cleaner and should be good enough for fdb.

Reply:
We have `basename(path)` to extract the filename part without the folders.

```suggestion
		if (basename(filename).find(testingStoragePrefix.toString()) == 0) {
```